File: wiz-event-mall-java/src/main/java/com/example/wizeventmalljava/domain/event/entity/EventWinner.java
package com.example.wizeventmalljava.domain.event.entity;

import com.example.wizeventmalljava.common.BaseEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class EventWinner extends BaseEntity {

	@Id @GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "event_winner_id")
	private Long id;

	@JoinColumn(name = "event_id", referencedColumnName = "event_id")
	@ManyToOne(fetch = FetchType.LAZY)
	private Event event;

	private Long memberId;

	@Builder
	public EventWinner(Long id, Event event, Long memberId) {
		this.id = id;
		this.event = event;
		this.memberId = memberId;
	}
}

File: wiz-event-mall/src/main/kotlin/com/example/wizeventmall/domain/event/entity/EventWinner.kt
package com.example.wizeventmall.domain.event.entity

import com.example.wizeventmall.common.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "event_winner")
class EventWinner(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "event_winner_id")
    var id: Long? = null,

    @JoinColumn(name = "event_id", referencedColumnName = "event_id")
    @ManyToOne(fetch = FetchType.LAZY)
    val event: Event,

    val memberId: Long

): BaseEntity() {
}